File: NaviTest/NaviTest/PinkieNavigation/NavigationItem/BarButtonItem/PinkieItemFactory.m
//
//  PinkieItemFactory.m
//  NaviTest
//
//  Created by jiazhiyi on 2019/4/9.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

#import "PinkieItemFactory.h"
#import "PinkieImageBarButtonItem.h"
#import "PinkieTitleBarButtonItem.h"
#import "PinkieDoubleBarButtonItem.h"

@implementation PinkieItemFactory

+ (id<PinkieBarButtonItem>)barButtonItemWithImage:(NSString *)image actionBlock:(BarItemBlock)actionBlock
{
    return [[PinkieImageBarButtonItem alloc] initWithImageName:image block:actionBlock];
}

+ (id<PinkieBarButtonItem>)barButtonItemWithTitle:(NSString *)title actionBlock:(BarItemBlock)actionBlock
{
    return [[PinkieTitleBarButtonItem alloc] initWithTitle:title block:actionBlock];
}

+ (id<PinkieBarButtonItem>)barButtonItem:(NSString *)image title:(NSString *)title actionBlock:(BarItemBlock)actionBlock
{
    return [[PinkieDoubleBarButtonItem alloc] initWithImageName:image title:title block:actionBlock];
}

@end

File: CustomNavigationBar/CustomNavigationBar/PinkieNavigation/NavigationItem/BarButtonItem/PinkieItemFactory.swift
//
//  PinkieItemFactory.swift
//  CustomNavigationBar
//
//  Created by jiazhiyi on 2019/4/8.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

import Foundation

class PinkieItemFactory {
    static func barButtonItem(image: String, actionBlock: (() -> Void)?) -> PinkieBarButtonItem {
        return PinkieImageBarButtonItem(imageName: image, block: actionBlock)
    }
    
    static func barButtonItem(title: String, actionBlock: (() -> Void)?) -> PinkieBarButtonItem {
        return PinkieTitleBarButtonItem(title: title, block: actionBlock)
    }
    
    static func barButtonItem(image: String, title: String, actionBlock: (() -> Void)?) -> PinkieBarButtonItem {
        return PinkieDoubleBarButtonItem(imageName: image, title: title, block: actionBlock)
    }
}
